Pod/Classes/ResultType.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  ResultType.swift
+//  Pods
+//
+//  Created by Bueno McCartney on 12/7/16.
+//
+//
+
+import Foundation
+
+public protocol ResultType {
+    associatedtype Value
+    
+    init(_ value: Value?)
+    init(_ error: Error)
+    
+    var value: Value? { get }
+    var error: Error? { get }
+}
+
+public enum Result<T>: ResultType {
+    public typealias Value = T
+    
+    case success(Value?)
+    case failure(Error)
+    
+    public var value: Value? {
+        switch self {
+        case .success(let v): return v
+        default: return nil
+        }
+    }
+    
+    public var error: Error? {
+        switch self {
+        case .failure(let e): return e
+        default: return nil
+        }
+    }
+    
+    public init(_ error: Error) {
+        self = .failure(error)
+    }
+    
+    public init(_ value: Value?) {
+        if let v = value as? Error {
+            self = .failure(v)
+        }
+        else if let v = value {
+            self = .success(v)
+        }
+        else {
+            self = .success(nil)
+        }
+    }
+}
+
+
+public extension Result {
+    public var isEmpty: Bool {
+        return value == nil
+    }
+}
+
+// MARK: - Helper Handling -
+public extension Result {
+    func onSuccess(handler: (_ data: Value?) -> Void) {
```

Comment:
These were mostly for convenience instead of doing switch on result or doing `if let err = result.err`.  Given that generally you'd setup some kind of func `setupWithData(data)` you could pass that straight to the result.  `result.onSuccess = setupWithData`